topic4/contracts/interfaces/IMath.sol

```diff
@@ -1,15 +1,15 @@
 pragma solidity ^0.4.23;
 
 interface IMath {
-	function stateNumber() external view;
+	function stateNumber() external view returns (uint);
 
 	function resetStateNumber() external;
```

Comment:
10. Setting the functions visibility specifiers to external in the interface **does let you** to implement them as public in the contract. Keep in mind that there is significant difference b/n external and public visibility - the latter allows you to call the function from 'inside' of the same contract, while the former allows only external calls( unless you use `this.` , but that is not optimal).